### PR TITLE
Update and rename customavatar.js to avatarcommands.js

### PR DESCRIPTION
**Pull request by BlazingAura**

### CODE REVIEW

**Comment by TheFenderStory** [comment]
I think the customavatar command system we currently have works better, although it does have some support issues. Also its not needed for you to tell people they need to npm install request when we could simply require it in our dependencies.

**Comment by BlazingAura** [comment]
the current customavatar.js has issues ....

**Comment by TheFenderStory** [comment]
Care to elaborate. Let's see if we can improve it

**Comment by Irraquated** [comment]
Although it is a good idea, I do not think a confirming system is needed for obvious reasons.

File: chat-plugins/avatarcommands.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			self.sendReply('|html|Your new avatar has been set to-<br/><img src = "' + targetArray[1] + '" width = 80 height = 80>');
+			response.pipe(fs.createWriteStream('config/avatars/' + targetArray[0] + '.' + format));
+		});
+	},
+
+	removeavatar: function (target, room, user, connection, cmd) {
+		if (typeof user.avatar === 'Number') return this.sendReply('You do not own a custom avatar.');
+		if (toId(target) !== 'confirm')
+			return this.sendReply('WARNING: If you choose to delete your avatar now, it cannot be recovered later. If you\'re sure you want to do this, enter \'/removeavatar confirm.\'');
+		delete avatarlist[user.userid];
```
Comment (Irraquated) [inline chat-plugins/avatarcommands.js]: Although it is a good idea, I do not think a confirming system is needed for obvious reasons.

**Comment by BlazingAura** [comment]
well fender it basically downloads the respected avatar from the link and later adds it in customavatars.json. Only issue is you later gotta copy it to config to make the change permanent!

**Comment by SilverTactic** [comment]
Hi there~
Okay so about this pull request, I feel that the original one should be used and not this. Reason: I strongly believe that this avatar command is the one I created, judging from the similarity between this code and the one I made this April on another repo I was working on at that time
https://github.com/Yggdrasil-League/Yggdrasil/commit/4cfeac7e3bb0aa452ae734368840acfb08ad24c2
Oh also, the customavatars.json file stores users' avatars. I used it as a replacement for the object in config.js
https://github.com/Yggdrasil-League/Yggdrasil/commit/1a06d07aa9f6b91e223a081d9684a0f208829c65
It's something I made a long time ago (and something I discarded too, because I didn't like it).

Either way, I feel it shouldn't be used simply because it can be abused. The request statements (like 29 to 38) simply check if the webpage exists, and then checks for a 404 error. If neither errors are encountered, then the response received from the page is written onto an image file in config/avatars. The script doesn't check whether the inputted url is actually an image file or not. You could simply take a url which contains (say,) several GB of data and then add .png at the end of it. The server would still write the response received onto an image file.

**Comment by BlazingAura** [comment]
so it seems, tho I cleared stuff up with siiilver about that. Anyways as Siiiilver has stated use kota's custom avatar :)

**Comment by SilverTactic** [comment]
Yeah, I guess that's that then xD

**Comment by TheFenderStory** [comment]
I do like the idea of making a new custom avatar command. We just need to make one with uses a program like GIFscile to handle the images. Perhaps its something worth looking into.

**Comment by SilverTactic** [comment]
Maybe there's another way to check if something's an image

**Pull request closed without merging**